homeassistant/components/media_player/channels.py

```diff
@@ -0,0 +1,395 @@
+# pylint: disable=too-many-instance-attributes
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Channels platform."""
+
+    device = ChannelsApp(
+                config.get('name', 'Channels'),
+                config.get(CONF_HOST),
+                config.get(CONF_PORT, 57000)
+             )
+
+    if DATA_CHANNELS not in hass.data:
+        hass.data[DATA_CHANNELS] = []
+
+    add_devices([device])
+    hass.data[DATA_CHANNELS].append(device)
+
+    def service_handler(service):
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+
+        if entity_ids:
+            devices = [device for device in hass.data[DATA_CHANNELS]
+                       if device.entity_id in entity_ids]
+        else:
+            devices = hass.data[DATA_CHANNELS]
+
+        for device in devices:
+            if service.service == SERVICE_SEEK_FORWARD:
+                device.seek_forward()
+            elif service.service == SERVICE_SEEK_BACKWARD:
+                device.seek_backward()
+            elif service.service == SERVICE_SEEK_BY:
+                seconds = service.data.get('seconds')
+                device.seek_by(seconds)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SEEK_FORWARD, service_handler,
+        schema=CHANNELS_SCHEMA)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SEEK_BACKWARD, service_handler,
+        schema=CHANNELS_SCHEMA)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SEEK_BY, service_handler,
+        schema=CHANNELS_SEEK_BY_SCHEMA)
+
+
+class ChannelsApp(MediaPlayerDevice):
+    """Representation of a Channels instance."""
+
+    # pylint: disable=too-many-public-methods
+    def __init__(self, name, host, port):
+        """Initialize the Channels app."""
+        self._name = name
+        self._host = host
+        self._port = port
+
+        self.client = Channels(self._host, self._port)
+
+        self.status = None
+        self.muted = None
+
+        self.channel_number = None
+        self.channel_name = None
+        self.channel_image_url = None
+
+        self.now_playing_title = None
+        self.now_playing_episode_title = None
+        self.now_playing_season_number = None
+        self.now_playing_episode_number = None
+        self.now_playing_summary = None
+        self.now_playing_image_url = None
+
+        self.favorite_channels = []
+
+        self.update()
+
+    def update_favorite_channels(self):
+        self.favorite_channels = self.client.favorite_channels()
+
+    def update_state(self, state_hash):
+        """Update all the state properties with the passed in dictionary."""
+        self.status = state_hash.get('status', "stopped")
+        self.muted = state_hash.get('muted', False)
+
+        channel_hash = state_hash.get('channel', None)
+        np_hash = state_hash.get('now_playing', None)
+
+        if channel_hash:
+            self.channel_number = channel_hash.get('channel_number', None)
```

Comment:
`get()` 2nd parameter defaults to `None` when omitted so you can ditch all the `None`s if you want